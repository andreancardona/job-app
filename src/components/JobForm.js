import React, { Fragment }from 'react';
import '../styles/_job-form.scss';

const JobForm = (props) => {
  const handleInputChange = (event) => {
		const { name, value } = event.target

		props.setJob({ ...props.job, [name]: value })
	}

  // Update header and subHeader based current form
  const header = props.adding ? 'Add a new job' : 'Edit a job';

  const subHeader = props.adding ? 'Fill out the information for your new job listing.' : 'Edit the information for your job listing.';

  return (
    <Fragment>
      <div className="form-headers" tabIndex="0" role="heading" aria-level="1" aria-label={header + subHeader}>
        <div className="form-heading">
          {header}
        </div>
        <div className="form-sub-heading">
          {subHeader}
        </div>
      </div>
      <form className="job-form">
        <label className="label" tabIndex="0">{props.labels.title} *</label>
        <div className="sub-label" tabIndex="0">{props.subLabels.title}</div>
          <input className="input" tabIndex="0" type="text" name="title" value={props.job.title} onChange={handleInputChange} />
        <label className="label" tabIndex="0">{props.labels.location} *</label>
        <div className="sub-label" tabIndex="0">{props.subLabels.location}</div>
          <input className="input" type="text" name="location" value={props.job.location}tabIndex="0"onChange={handleInputChange} />
        <label className="label" tabIndex="0">{props.labels.sponsorship} *</label>
        <div className="sub-label" tabIndex="0">{props.subLabels.sponsorship}</div>    
          <select className="select-input" name="sponsorship" onChange={handleInputChange} value={props.job.sponsorship}>
            <option value="Sponsor">Sponsor</option>
            <option value="Free">Free</option>
          </select>
        <label className="label" tabIndex="0">{props.labels.status} *</label>
        <div className="sub-label" tabIndex="0">{props.subLabels.status}</div> 
          <select className="select-input" onChange={handleInputChange} name="status" value={props.job.status} tabIndex="0">
            <option value="Open">Open</option>
            <option value="Paused">Paused</option>
            <option value="Closed">Closed</option>
          </select>
      </form>
    </Fragment>  
  )
}

export default JobForm;