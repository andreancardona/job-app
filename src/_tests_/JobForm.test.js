import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import JobForm from '../components/JobForm';

 // Initial job
 const job = {
  job: { id: 1, title: 'Product Manager', location: 'Austin, TX', posted: '10/04/2020', sponsorship: 'Free', status: 'Open' },
}

describe('The <JobForm /> component', () => {
  test('JobForm renders with JobForm props', () => {
    const tree = renderer.create(<JobForm job={job} />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('Job should not fail any accessibility tests', async () => {
    const { container } = render(<JobForm />);
    expect(await axe(container)).toHaveNoViolations();
  });

});
