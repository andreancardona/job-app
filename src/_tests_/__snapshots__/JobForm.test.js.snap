// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <JobForm /> component JobForm renders with JobForm props 1`] = `
<div
  className="job-form-container"
>
  <div
    aria-label="Add new jobFill out the information for your new job listing."
    aria-level="1"
    className="form-headers"
    role="heading"
    tabIndex="0"
  >
    <div
      className="form-heading"
    >
      Add new job
    </div>
    <div
      className="form-sub-heading"
    >
      Fill out the information for your new job listing.
    </div>
  </div>
  <form
    className="job-form"
  >
    <label
      className="label"
      tabIndex="0"
    >
      Job title
       *
    </label>
    <div
      aria-label="What is the name of the role?"
      className="sub-label"
      tabIndex="0"
    >
      What is the name of the role?
    </div>
    <input
      aria-label="title"
      className="input"
      name="title"
      onChange={[Function]}
      tabIndex="0"
      type="text"
      value=""
    />
    <label
      className="label"
      tabIndex="0"
    >
      Location
       *
    </label>
    <div
      aria-label="Where is this job?"
      className="sub-label"
      tabIndex="0"
    >
      Where is this job?
    </div>
    <input
      aria-label="title"
      className="input"
      name="location"
      onChange={[Function]}
      tabIndex="0"
      type="text"
      value=""
    />
    <label
      aria-label="Do you want to promote this job?"
      className="label"
      tabIndex="0"
    >
      Sponsorship
       *
    </label>
    <div
      aria-label="Do you want to promote this job?"
      className="sub-label"
      tabIndex="0"
    >
      Do you want to promote this job?
    </div>
    <select
      aria-label="sponsorship options"
      className="select-input"
      name="sponsorship"
      onChange={[Function]}
      value="Sponsor"
    >
      <option
        value="Sponsor"
      >
        Sponsor
      </option>
      <option
        value="Free"
      >
        Free
      </option>
    </select>
    <label
      aria-label="Are you ready to make this job posting public?"
      className="label"
      tabIndex="0"
    >
      Status
       *
    </label>
    <div
      aria-label="Are you ready to make this job posting public?"
      className="sub-label"
      tabIndex="0"
    >
      Are you ready to make this job posting public?
    </div>
    <select
      aria-label="sponsorship options"
      className="select-input"
      name="status"
      onChange={[Function]}
      tabIndex="0"
      value="Open"
    >
      <option
        value="Open"
      >
        Open
      </option>
      <option
        value="Paused"
      >
        Paused
      </option>
      <option
        value="Closed"
      >
        Closed
      </option>
    </select>
  </form>
  <div
    className="job-form--footer"
  >
    <div
      className="footer-buttons"
    >
      <button
        className="cancel-button"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="add-save-button"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
